doc/kubebuilder-migration/project_migration_guide.md

```diff
@@ -0,0 +1,374 @@
+		}
+	}
+	return ctrl.Result{}, nil
+}
+```
+
+## Generate CRDs
+
+Run [`make manifests`][generate_crd] to generate CRD manifests. They would be generated inside the `config/` folder.
+
+```sh
+$ make manifests
+go: creating new go.mod: module tmp
+go: finding sigs.k8s.io v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd/controller-gen v0.2.5
+/Users/admin/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+```
+
+## Operator Manifests
+
+### Operator deployment manifests
+
+The operator deployment manifest [`operator.yaml`][deployment_yaml] present inside `deploy/` folder in SDK project can be copied to `config/manager/manager.yaml`
+
+**Note**: 
+The kustomize file requires the operator deployment manifest to have the field `namespace` which is missing in the current operator.yaml of sdk project.
+
+### RBAC permissions
+
+The RBAC related yaml files present in `config/rbac/` of kubebuilder project are generated through the [rbac markers][rbac_markers] present in `controller/memcached_controller.go`.
+
+In our example of memcached operator, add the following rbac markers in `memcached_controller.go`:
+
+```Go
+// +kubebuilder:rbac:groups=apps,resources=deployments;pods;daemonsets;replicasets;statefulsets,verbs=get;update;patch;list;create;delete;watch
+```
+**Note**:
+To update `config/rbac/role.yaml` after changing the markers, run `make manifests`.
+
+## Build and run the operator:
+
+Before running the operator, run `make install` to apply the crds in the cluster.
+
+```sh
+$ make install
+go: creating new go.mod: module tmp
+go: finding sigs.k8s.io/controller-tools/cmd v0.2.5
+go: finding sigs.k8s.io v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd/controller-gen v0.2.5
+/Users/vnarsing/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
+kustomize build config/crd | kubectl apply -f -
+customresourcedefinition.apiextensions.k8s.io/memcacheds.cache.example.com configured
+```
+
+There are two ways to run the operator:
+- As Go program outside a cluster
+- As a Deployment inside a Kubernetes cluster
+
+### 1. Run locally outside the cluster
+
+To run the operator locally execute the following command:
+
+```
+make run ENABLE_WEBHOOKS=false
+```
+
+```sh
+$ make run ENABLE_WEBHOOKS=false
+go: creating new go.mod: module tmp
+go: finding sigs.k8s.io v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd v0.2.5
+go: finding sigs.k8s.io/controller-tools/cmd/controller-gen v0.2.5
+/Users/admin/go/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
+go fmt ./...
+go vet ./...
+```
+
+### 2. Run as a Deployment inside the cluster
+
+Build the operator image and push it to a registry.
+
+```sh
+$ make docker-build docker-push IMG=<some-registry>/<project-name>:tag
+```
```

Comment:
Another option is to set `IMG` in the Makefile, right?

Reply:
Yes, Included in the note.